governance/program/src/instruction.rs

```diff
@@ -0,0 +1,264 @@
+    ///   0. `[writable]` Transaction account you wish to execute
+    ///   1. `[writable]` Proposal State account
+    ///   2. `[]` Program being invoked account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Governance account
+    ///   5. `[]` Governance program account pub key
+    ///   6. `[]` Clock sysvar
+    ///   7+ Any extra accounts that are part of the instruction, in order
+    Execute,
+
+    /// Creates an empty governance vote record
+    ///
+    ///   0. `[]` Governance Vote Record account. Needs to be set with pubkey set to PDA with seeds of the
+    ///           program account key, proposal key, your voting account key
+    ///   1. `[]` Proposal key
+    ///   2. `[]` Your Vote account
+    ///   3. `[]` Payer
+    ///   4. `[]` System account.
+    CreateEmptyGovernanceVoteRecord,
+
+    /// [Requires tokens of the Governance Mint or Council Mint depending on type of Proposal]
+    /// Deposits vote tokens to be used during the voting process on a Proposal.
+    /// These tokens are removed from your account and can be returned by withdrawing
+    /// them from the Proposal (but then you will miss the vote.)
+    ///
+    ///   0. `[writable]` Governance Vote Record account. See Vote docs for more detail
+    ///   1. `[writable]` Initialized Vote account to hold your received voting tokens
+    ///   2. `[writable]` User Source Token account to deposit tokens from
+    ///   3. `[writable]` Source Token Holding account for Proposal that will accept the tokens in escrow
+    ///   4. `[writable]` Vote Mint account
+    ///   5. `[]` Proposal account.
+    ///   6. `[]` Transfer authority
+    ///   7. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   8. `[]` Token program account
+    DepositSourceTokens {
+        /// How many voting tokens to deposit
+        voting_token_amount: u64,
+    },
+
+    /// [Requires voting tokens]
+    /// Withdraws voting tokens.
+    ///
+    ///   0. `[writable]` Governance Vote Record account. See Vote docs for more detail
+    ///   1. `[writable]` Initialized Vote account from which to remove your voting tokens
+    ///   2. `[writable]` Initialized Yes Vote account from which to remove your voting tokens
+    ///   3. `[writable]` Initialized No Vote account from which to remove your voting tokens
+    ///   4. `[writable]` User Source Token account that you wish your actual tokens to be returned to
+    ///   5. `[writable]` Source Token Holding account owned by the Governance that will has the actual tokens in escrow
+    ///   6. `[writable]` Vote Mint account
+    ///   7. `[writable]` Yes Vote Mint account
+    ///   8. `[writable]` No Vote Mint account
+    ///   9. `[]` Proposal State account
+    ///   10. `[]` Proposal account
+    ///   11. `[]` Transfer authority
+    ///   12. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   13. `[]` Token program account
+    WithdrawVotingTokens {
```

Comment:
So this would be `WithdrawSourceTokens` or `WithdrawGovernanceTokens` for people to retract their vote?

Reply:
Correct. WithdrawSourceTokens it is.